hw/dv/dpi/jtagdpi/jtagdpi.c

```diff
@@ -294,17 +111,23 @@ void *jtagdpi_create(const char *display_name, int listen_port) {
       (struct jtagdpi_ctx *)calloc(1, sizeof(struct jtagdpi_ctx));
   assert(ctx);
 
-  ctx->listen_port = listen_port;
-  ctx->display_name = strdup(display_name);
-  assert(ctx->display_name);
+  // Set up socket details
+  ctx->sock.socket_run = true;
+  ctx->sock.listen_port = listen_port;
+  ctx->sock.display_name = strdup(display_name);
+  assert(ctx->sock.display_name);
+  ctx->sock.buf_in = NULL;
+  ctx->sock.buf_out = NULL;
```

Comment:
Adding an init function that encapsulates these lines would probably make it slightly safer to use.